### PR TITLE
Port implementation from loguru PR

### PR DESCRIPTION
**Pull request by kornicameister**
Original PR: https://github.com/Delgan/loguru/pull/276

### CODE REVIEW

**Comment by kornicameister** [comment]
@Delgan I will move include corner cases in this or in another PR. I'd like to get your opinion on this PR first if you have time and don't mind :)

**Comment by kornicameister** [comment]
```
from loguru import logger

def foo(a, b):
    a / b

try:
    foo(1, 0)
except Exception:
    logger.exception("An error occurred while executing: '{}'", foo)
```

Ok, I understand why this shouldn't be evaluated as an error. Point made, but this actually completes things. Correct me, but would doing `logger.opt(lazy=True).exception(...)` actually tried to call `foo`?

File: loguru_mypy/__init__.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+            f'{extra_kwarg_name} keyword argument not found in log message',
+            context=log_msg_expr,
+            code=ERROR_BAD_KWARG,
+        )
+
+    return ctx.default_return_type
+
+
+class LoguruPlugin(Plugin):
+    builtin_severities = (
```
Comment (Delgan) [inline loguru_mypy/__init__.py:130]: Maybe add `trace` and `success` too (those are valid logging methods)?

**Comment by Delgan** [comment]
> Ok, I understand why this shouldn't be evaluated as an error. Point made, but this actually complicates things. Correct me, but would doing `logger.opt(lazy=True).exception(...)` actually tried to call `foo`?

Indeed, there are a lot of special cases that can make things complicated. :grin:

I'm not sure fixing this worth the trouble. It was just to give an example of the limitations we were going to face. There exist probably other edge cases of that kind, unfortunately.

A false positive is more critical than a false negative, though.

I don't know how this can be corrected. Surely we should introspect the internal attributes of the logger, if that is possible.

Your description of the `lazy` parameter is absolutely correct. If set to `True`, the logging arguments will be called and returned value will be used for the formatting. If set to `False` (the default), the argument could be of any type and will be formatted as is (including a function which will not be called in this case).

File: loguru_mypy/__init__.py (hunk 1 above)
Comment (kornicameister) [reply loguru_mypy/__init__.py:130]: Done

**Comment by kornicameister** [comment]
@Delgan I've created an issue #6 to approach `lazy={true,false}` later on.

**Pull request merged**